src/resolvers/resolveUrls.js

```diff
@@ -22,23 +23,26 @@ async function resolvedUrls(root, { urls }) {
         }
 
         const fetchResult = await fetcher;
-        return {
+        call.write({
           ...fetchResult,
+          top_image_url: fetchResult.topImageUrl,
           url,
-        };
+          successfully_resolved: true,
+        });
       } catch (e) {
         // eslint-disable-next-line no-console
         console.error('[resolvedUrls]', url, e);
-
+        let errMsg;
         if (e instanceof ResolveError) {
-          return {
-            error: e.returnedError,
-            url,
-          };
+          errMsg = e.returnedError;
         }
+        call.write({
```

Comment:
@changhc 
Does this mean that if I call `resolveUrls('http://a.com', 'http://b.com')`,
It's possible that I may get `[{...b's result}, {...a's result}]`, if `b.com` loads faster?

Reply:
Yes, but this is the characteristic of stream responses. If one is done earlier, it can be returned earlier and doesn't need to wait for others. In our use case, even if our client still wait until all responses are returned before proceeding, it can save some transmission time since results can be returned in batches when the server is still working on other requests.

Reply:
Good point. We should handle the ordering issue at the application (rumors-api) side.